Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -56,25 +65,159 @@ namespace WebCore {
 
 WTF_MAKE_ISO_ALLOCATED_IMPL(CookieStore);
 
-Ref<CookieStore> CookieStore::create(Document* document)
+class CookieStore::MainThreadBridge : public ThreadSafeRefCounted<MainThreadBridge, WTF::DestructionThread::Main> {
+public:
+    static Ref<MainThreadBridge> create(CookieStore& cookieStore)
+    {
+        return adoptRef(*new MainThreadBridge(cookieStore));
+    }
+
+    void get(CookieStoreGetOptions&&, Function<void(CookieStore&, ExceptionOr<Vector<Cookie>>&&)>&&);
+    void getAll(CookieStoreGetOptions&&, URL&&, Function<void(CookieStore&, ExceptionOr<Vector<Cookie>>&&)>&&);
+    void set(CookieInit&& options, Cookie&&, Function<void(CookieStore&, std::optional<Exception>&&)>&&);
+
+private:
+    explicit MainThreadBridge(CookieStore&);
+
+    void ensureOnMainThread(Function<void(ScriptExecutionContext&)>&&);
+    void ensureOnContextThread(Function<void(CookieStore&)>&&);
+
+    WeakPtr<CookieStore, WeakPtrImplWithEventTargetData> m_cookieStore;
+    ScriptExecutionContextIdentifier m_contextIdentifier;
+};
+
+CookieStore::MainThreadBridge::MainThreadBridge(CookieStore& cookieStore)
+    : m_cookieStore(cookieStore)
+    , m_contextIdentifier(cookieStore.scriptExecutionContext()->identifier())
```

Comment:
Missing null check for the context again..

Reply:
Ah sorry. Fixed!